packages/context/src/resolver.ts

```diff
@@ -133,3 +133,16 @@ export function resolveInjectedProperties(fn: Function, ctx: Context): KV | Prom
   }
 }
 
+/**
+ * resolve a ValueOrPromise<T> to T
+ */
+export async function resolveValueOrPromise<T>(instanceOrPromise: ValueOrPromise<T>): Promise<T> {
+  if (isPromise(instanceOrPromise)) {
+    const providerPromise = instanceOrPromise as Promise<T>;
+    const instance: T = await providerPromise;
+    return instance;
+  } else {
+    const instance: T = instanceOrPromise as T;
+    return instance;
+  }
```

Comment:
FWIW, this whole function can be shortened to a single line.

```ts
export async function resolveValueOrPromise<T>(instanceOrPromise: ValueOrPromise<T>): Promise<T> {
  return await Promise.resolve(instanceOrPromise);
}
```

And inlined at call site:

```diff
- return resolveValueOrPromise<Provider<T>>(providerOrPromise);
+ return Promise.resolve(providerOrPromise);
```